gridmet.py

```diff
@@ -0,0 +1,152 @@
+    :param str end_date: End date of data collection yyyy-mm-dd
+    :param str lon_min: bbox of aoi longitude min
+    :param str lon_max: bbox of aoi longitude max
+    :param str lat_min: bbox of aoi latitude min
+    :param str lat_max: bbox of aoi latitude max
+    :param str g2s_file_prefix: prefix used to save the grd2shp_xarray
+    :return: g2s array
+    """
+
+    ## Create dictionary of short variable names with official long variable name
+    data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+                          'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+                          'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+                          'daily_minimum_relative_humidity', 'daily_mean_specific_humidity']
+
+    all_data_vars_dict = dict(zip(data_vars_shrt_all, data_vars_long_all))
+    subset = {key: all_data_vars_dict[key] for key in var_short}
+    print(subset)
+    del all_data_vars_dict
+
+    ## geospatial vector file
+    gdf = gpd.read_file(polygon_file_path)
+
+    ## check if only 1 var (short and long) is provided, change to list
+    if type(var_short) == str:
+        var_short = [var_short]
+        print(var_short)
+    ## check that var_short is a list
+    elif type(var_short) != list:
+        var_short = [var_short]
+        print(var_short)
+
+    ## empty list for the xarray.datasets
+    data_xr_ds_list = []
+
+    ## Creating ds
+    for i in var_short:
+        # Get url:
+        print(i)
+        ## source: http://thredds.northwestknowledge.net:8080/thredds/reacch_climate_MET_aggregated_catalog.html
+        url = 'http://thredds.northwestknowledge.net:8080/thredds/dodsC/agg_met_{}_1979_CurrentYear_CONUS.nc'.format(i)
+        print(url)
+
+        # call data from url
+        start = time.perf_counter()
+        df = xr.open_dataset(url + '#fillmismatch')
+
+        # Subset to timeframe and bbox of interest:
+        df_subset = df.sel(day=slice(start_date, end_date),
+                           lon=slice(lon_min, lon_max),
+                           lat=slice(lat_max, lat_min))
+        end = time.perf_counter()
+        print('finish in {} seconds'.format(round(start - end), 2))
+
+        # Append to list of xr.datasets
+        data_xr_ds_list.append(df_subset)
+
+    ## Creating dictionary for next process
+    data_dict = dict(zip(var_short, data_xr_ds_list))
+    ## taking first dict item for creation of weight's file.
+    wghtmap_var = list(data_dict.keys())[0]
+
+    ## Produce weightmap
+    start = time.perf_counter()
+    weightmap = xa.pixel_overlaps(data_dict[wghtmap_var], gdf)
+    end = time.perf_counter()
+    print(f'finished agg in {round(end - start, 2)} second(s)')
+
+    with open(output_data_folder + 'grd2shp_weights_{}.pickle'.format(wghtmap_var), 'wb') as file:
```

Comment:
I think it's a good idea to make a weight file variable. That way you can use it in the `.initialize` function call too. Also, I think it's safer to use `os.path.join` instead of `+` so you don't have to worry about whether to include the `/` or not. And since this weightmap is consistent between variables, I think it makes sense to leave that out of the filename.

```suggestion
    wght_file = os.path.join(output_data_folder, 'grd2shp_weights.pickle')
    with open(wght_file, 'wb') as file:
```